src/documents/file_handling.py

```diff
@@ -194,33 +196,44 @@ def generate_filename(
             # Convert UTC database datetime to localized date
             local_added = timezone.localdate(doc.added)
             local_created = timezone.localdate(doc.created)
+            placeholders = {
+                "title": pathvalidate.sanitize_filename(
+                    doc.title,
+                    replacement_text="-",
+                ),
+                "correspondent": correspondent,
+                "document_type": document_type,
+                "created": local_created.isoformat(),
+                "created_year": local_created.strftime("%Y"),
+                "created_year_short": local_created.strftime("%y"),
+                "created_month": local_created.strftime("%m"),
+                "created_month_name": local_created.strftime("%B"),
+                "created_month_name_short": local_created.strftime("%b"),
+                "created_day": local_created.strftime("%d"),
+                "added": local_added.isoformat(),
+                "added_year": local_added.strftime("%Y"),
+                "added_year_short": local_added.strftime("%y"),
+                "added_month": local_added.strftime("%m"),
+                "added_month_name": local_added.strftime("%B"),
+                "added_month_name_short": local_added.strftime("%b"),
+                "added_day": local_added.strftime("%d"),
+                "asn": asn,
+                "tags": tags,
+                "tag_list": tag_list,
+                "owner_username": owner_username_str,
+                "original_name": original_name,
+                "doc_pk": f"{doc.pk:07}",
+            }
+
+            for field in CustomField.objects.all():
```

Comment:
That's expected. 
Given a document wich has customfield1 filled only but uses a storage path with customfield2 as placeholder. Then the format function would break because a placeholder is missing.
Better way in my opinion: Set all custom fields to "no_value_default" to prevent this issue and the take all values related to the document.

If there's a way to do this without this two lookups, let me know.